spring-integration-core/src/main/java/org/springframework/integration/dsl/RecipientListRouterSpec.java

```diff
@@ -89,8 +97,14 @@ public RecipientListRouterSpec recipientMessageSelector(String channelName, Mess
 	 * @return the router spec.
 	 */
 	public <P> RecipientListRouterSpec recipient(String channelName, GenericSelector<P> selector) {
-		Assert.hasText(channelName);
-		((DslRecipientListRouter) this.target).add(channelName, selector);
+		MessageSelector messageSelector;
+		if (selector instanceof MessageSelector) {
+			messageSelector = (MessageSelector) selector;
+		}
+		else {
+			messageSelector = new MethodInvokingSelector(new LambdaMessageProcessor(selector, null));
```

Comment:
It might be called `Lamba` by accident, but the purpose is still there single method invocation purpose for provided `payloadType`. My idea to rely on the  strict `GenericSelector` implementation and bypass `MessagingMethodInvokerHelper` which is based on the SpEL and a bit slower than direct method invocation.